include/exadg/postprocessor/pointwise_output_generator_base.cpp

```diff
@@ -68,52 +59,26 @@ template class PointwiseOutputDataBase<3>;
 template<int dim, typename Number>
 void
 PointwiseOutputGeneratorBase<dim, Number>::evaluate(VectorType const & solution,
-                                                    double const &     time,
-                                                    int const &        time_step_number)
+                                                    double const       time,
+                                                    bool const         unsteady)
 {
-  if(pointwise_output_data.write_output && pointwise_output_data.evaluation_points.size() > 0)
-  {
-    if(Utilities::is_unsteady_timestep(time_step_number))
-    {
-      // small number which is much smaller than the time step size
-      double const EPSILON = 1.0e-10;
-
-      // In the first time step, the current time might be larger than start_time. In that
-      // case, we first have to reset the counter in order to avoid that output is written every
-      // time step.
-      if(reset_counter)
-      {
-        counter += int((time - pointwise_output_data.start_time + EPSILON) /
-                       pointwise_output_data.interval_time);
-        reset_counter = false;
-      }
-
-
-      if((time > (pointwise_output_data.start_time + counter * pointwise_output_data.interval_time -
-                  EPSILON)) &&
-         time < pointwise_output_data.end_time + EPSILON)
-      {
-        if(pointwise_output_data.update_points_before_evaluation)
-          reinit_remote_evaluator();
-
-        write_time(time);
-        do_evaluate(solution);
-
-        ++counter;
-      }
-    }
-    else
-    {
-      write_time(time);
-      do_evaluate(solution);
-      ++counter;
-    }
-  }
+  AssertThrow(unsteady, dealii::ExcMessage("Only implemented for the unsteady case."));
+
+  AssertThrow(time_control.is_restarted() == false,
+              dealii::ExcMessage(
+                "Only implemented in the case that the simulation is not restarted"));
```

Comment:
To get rid of `is_restarted()`, check that `time_control.get_counter() == 0` in the first time step of a simulation.